src/TesApi.Web/BatchScheduler.cs

```diff
@@ -406,17 +405,9 @@ public void ClearBatchLogState()
         private static string GetCromwellExecutionDirectoryPath(TesTask task)
             => GetParentPath(task.Inputs?.FirstOrDefault(IsCromwellCommandScript)?.Path.TrimStart('/'));
 
-        private static string GetBatchExecutionDirectoryPath(TesTask task)
+        private static string GetStorageUploadPath(TesTask task, string storageAccountName)
         {
-            var cromwellDir = GetCromwellExecutionDirectoryPath(task);
-            if (cromwellDir is not null)
-            {
-                return $"{cromwellDir}/{BatchExecutionDirectoryName}";
-            }
-            else
-            {
-                return $"{ExecutionsPathPrefix.TrimStart('/')}";
-            }
+            return $"{storageAccountName}/{TesExecutionsPathPrefix.TrimStart('/')}/{task.Id}";
```

Comment:
instead of: /{TesExecutionsPathPrefix.TrimStart('/')}

more performant to do:

{TesExecutionsPathPrefix}